app/javascript/components/lease_ups/LeaseUpApplicationsPage.js

```diff
@@ -156,29 +157,42 @@ const LeaseUpApplicationsPage = () => {
     loadPage(0, filters)
   }
 
-  const handleCreateStatusUpdate = async ({ applicationId, comment, status, subStatus }) => {
-    const { applications } = state
-
-    createFieldUpdateComment(applicationId, status, comment, subStatus)
-      .then((response) => {
-        each(applications, (app, index) => {
-          if (app.application_id === applicationId) {
-            updateResults(`[${index}]['lease_up_status']`, status)
-            updateResults(
-              `[${index}]['status_last_updated']`,
-              moment().format(SALESFORCE_DATE_FORMAT)
-            )
-            updateResults(`[${index}]['sub_status']`, subStatus)
-          }
-        })
+  const handleStatusModalSubmit = async (submittedValues) => {
+    updateStatusModal({ loading: true })
+    const { applicationIds, isBulkChange } = statusModalState
+    const { status, subStatus } = submittedValues
+    var comment = submittedValues.comment && submittedValues.comment.trim()
+    const data = {
+      applicationIds,
+      status,
+      comment,
+      ...(subStatus ? { subStatus } : {}),
+      isBulkChange: isBulkChange
+    }
 
+    createStatusUpdates(data)
+  }
+  const createStatusUpdates = async ({
+    applicationIds,
+    comment,
+    status,
+    subStatus,
+    isBulkChange
+  }) => {
+    const statusUpdateRequests = applicationIds.map((appId) =>
+      createFieldUpdateComment(appId, status, comment, subStatus)
+    )
+    Promise.all(statusUpdateRequests)
```

Comment:
if this is an async function you should return this promise
```suggestion
    return Promise.all(statusUpdateRequests)
```